diff --git a/node_modules/@convex-dev/r2/dist/esm/client/index.js b/node_modules/@convex-dev/r2/dist/esm/client/index.js
index d0571ff..0b0f66d 100644
--- a/node_modules/@convex-dev/r2/dist/esm/client/index.js
+++ b/node_modules/@convex-dev/r2/dist/esm/client/index.js
@@ -12,6 +12,7 @@ const parseConfig = (config) => {
         endpoint: "R2_ENDPOINT",
         accessKeyId: "R2_ACCESS_KEY_ID",
         secretAccessKey: "R2_SECRET_ACCESS_KEY",
+        forcePathStyle: "R2_FORCE_PATH_STYLE",
     };
     const missingEnvVars = Object.keys(configVars).filter((key) => !config[key]);
     if (missingEnvVars.length > 0) {
@@ -75,6 +76,7 @@ export class R2 {
             endpoint: options?.R2_ENDPOINT ?? process.env.R2_ENDPOINT,
             accessKeyId: options?.R2_ACCESS_KEY_ID ?? process.env.R2_ACCESS_KEY_ID,
             secretAccessKey: options?.R2_SECRET_ACCESS_KEY ?? process.env.R2_SECRET_ACCESS_KEY,
+            forcePathStyle: options?.R2_FORCE_PATH_STYLE ?? process.env.R2_FORCE_PATH_STYLE,
         };
         this.r2 = createR2Client(parseConfig(this.config));
     }
diff --git a/node_modules/@convex-dev/r2/dist/esm/shared.js b/node_modules/@convex-dev/r2/dist/esm/shared.js
index 8837863..cbbdf07 100644
--- a/node_modules/@convex-dev/r2/dist/esm/shared.js
+++ b/node_modules/@convex-dev/r2/dist/esm/shared.js
@@ -5,6 +5,7 @@ export const r2ConfigValidator = v.object({
     endpoint: v.string(),
     accessKeyId: v.string(),
     secretAccessKey: v.string(),
+    forcePathStyle: v.boolean(),
 });
 export const createR2Client = (args) => {
     return new S3Client({
@@ -14,6 +15,7 @@ export const createR2Client = (args) => {
             accessKeyId: args.accessKeyId,
             secretAccessKey: args.secretAccessKey,
         },
+        ...(args.forcePathStyle ? { forcePathStyle: true } : {}),
     });
 };
 export const withoutSystemFields = (fields) => {
diff --git a/node_modules/@convex-dev/r2/src/client/index.ts b/node_modules/@convex-dev/r2/src/client/index.ts
index 83aeaea..8fcd93e 100644
--- a/node_modules/@convex-dev/r2/src/client/index.ts
+++ b/node_modules/@convex-dev/r2/src/client/index.ts
@@ -40,6 +40,7 @@ const parseConfig = (config: Infer<typeof r2ConfigValidator>) => {
     endpoint: "R2_ENDPOINT",
     accessKeyId: "R2_ACCESS_KEY_ID",
     secretAccessKey: "R2_SECRET_ACCESS_KEY",
+    forcePathStyle: "R2_FORCE_PATH_STYLE",
   };
   const missingEnvVars = Object.keys(configVars).filter(
     (key) => !config[key as keyof typeof config]
@@ -129,6 +130,7 @@ export class R2 {
       R2_ACCESS_KEY_ID?: string;
       R2_SECRET_ACCESS_KEY?: string;
       defaultBatchSize?: number;
+      forcePathStyle?: boolean;
     } = {}
   ) {
     this.config = {
@@ -137,6 +139,7 @@ export class R2 {
       accessKeyId: options?.R2_ACCESS_KEY_ID ?? process.env.R2_ACCESS_KEY_ID!,
       secretAccessKey:
         options?.R2_SECRET_ACCESS_KEY ?? process.env.R2_SECRET_ACCESS_KEY!,
+      forcePathStyle: options?.forcePathStyle ?? false,
     };
     this.r2 = createR2Client(parseConfig(this.config));
   }
diff --git a/node_modules/@convex-dev/r2/src/shared.ts b/node_modules/@convex-dev/r2/src/shared.ts
index 4165bf6..758c5d5 100644
--- a/node_modules/@convex-dev/r2/src/shared.ts
+++ b/node_modules/@convex-dev/r2/src/shared.ts
@@ -7,9 +7,11 @@ export const r2ConfigValidator = v.object({
   endpoint: v.string(),
   accessKeyId: v.string(),
   secretAccessKey: v.string(),
+  forcePathStyle: v.string(),
 });
 
 export const createR2Client = (args: Infer<typeof r2ConfigValidator>) => {
+  console.log("args", args)
   return new S3Client({
     region: "auto",
     endpoint: args.endpoint,
@@ -17,7 +19,8 @@ export const createR2Client = (args: Infer<typeof r2ConfigValidator>) => {
       accessKeyId: args.accessKeyId,
       secretAccessKey: args.secretAccessKey,
     },
-  });
+    ...(args.forcePathStyle === "true" ? { forcePathStyle: true } : {}),
+    });
 };
 
 export const withoutSystemFields = <T extends Doc<TableNames>>(fields: T) => {
